src/WrapperApp/components/Simulation/SimulationProgressBar.tsx

```diff
@@ -1,3 +1,70 @@
-export function SimulationProgressBar() {
-	return <></>;
+import { Box, LinearProgress, Divider, Tooltip } from '@mui/material';
+import { TaskStatusData, TaskTime, StatusState } from '../../../services/ResponseTypes';
+import { useRef, useCallback, useEffect } from 'react';
+import Countdown from 'react-countdown';
+
+const getDateFromEstimation = (estimated?: TaskTime) => {
+	if (!estimated) return undefined;
+	const { hours, minutes, seconds } = estimated;
+	const date =
+		Date.now() +
+		(parseInt(hours) * 60 * 60 + parseInt(minutes) * 60 + parseInt(seconds)) * 1000;
+	return date;
+};
+
+type SimulationProgressBarProps = {
+	status: TaskStatusData;
+};
+const statusToColor = (status: StatusState) => {
+	switch (status) {
+		case StatusState.FAILED:
+			return 'error';
+		case StatusState.COMPLETED:
+			return 'success';
+		default:
+			return 'secondary';
+	}
+};
+
+export function SimulationProgressBar({ status }: SimulationProgressBarProps) {
+	const updateProgress = useCallback(() => {
+		const progress = (status?.simulatedPrimaries ?? 0) / (status?.requestedPrimaries ?? 1);
```

Comment:
With requested and simulated primaries there may be an issue coming soon.
Take a look here https://github.com/yaptide/yaptide/issues/227

In fact when requesting 100000 primaries on 100 cores, each core should simulate only 1000 primaries. That means it will send back status with primaries simulated from 0 till 1000.
That means calculation of progress is complicated.

Reply:
@p1003 told me that the requested amount is defined for each task separately. 
Right now is constant across all tasks and equal to the parameter given on the simulation run, but could be expanded to chunks calculated the way you described.

Reply:
Will be resolved in backend